lib/ash/filter/filter.ex

```diff
@@ -2854,11 +2851,13 @@ defmodule Ash.Filter do
             {:error, error}
         end
 
-      op_module = get_operator(field) && match?([_, _ | _], nested_statement) ->
-        with {:ok, [left, right]} <- hydrate_refs(nested_statement, context),
-             refs <- list_refs([left, right]),
+      (op_module = get_operator(field)) && match?([_, _ | _], nested_statement) ->
+        with {:ok, refs} <-
+               hydrate_refs(nested_statement, context),
+             refs <- list_refs(refs),
              :ok <- validate_refs(refs, context.root_resource, {field, nested_statement}),
-             {:ok, operator} <- Operator.new(op_module, left, right) do
+             {:ok, operator} <-
+               Operator.new(op_module, context.root_resource, refs) do
```

Comment:
This doesn't seem right to me. The left side of the operator shouldn't be the root resource. It should be the left/right ref

Reply:
I tested with the in operator: `[author1: [in: [:id1, :id2]]]`

The existing code turned `[:id1, :id2]` into `[left, right]`. I believe left should be `author1.id` and right `[:id1, :id2]`.

Reply:
Even if that is the case, the `left` side of the `in` operator wouldn't be a resource, it would be a field or some other expression. In this case, I'm pretty sure this is to handle something in the format of `[in: [:id, [1, 2, 3]]]`.

Reply:
Do we support both?

1. `[id: [in: [1, 2, 3]]]`
2. `[in: [id: [1, 2, 3]]]`

Reply:
🤔 I *think* so? If we add a test for both, it should highlight the issue, I think. I believe the second one will fail now because of the change to `Operator.new` and that the first one will work because it doesn't go through this code path.

Reply:
Yes, this fails now:

```elixir
      # prefix operator
      assert Filter.parse!(Post, author1: [in: [id: [id1, id2]]])
```

That makes it even harder to get this change working 😭

Reply:
The fix to `op_module` was required though, I can see how that would evaluate `op_module` to `true`.

Reply:
Alternatively, since this code path was broken already, we could just remove this match entirely :)